serve.go

```diff
@@ -721,3 +727,22 @@ func debugHTTPmiddleware(next http.Handler) http.Handler {
 		next.ServeHTTP(w, r)
 	})
 }
+
+func basicAuth(password string) string {
```

Comment:
I think I'm missing where this function is used? It looks like you've used the http `SetAuthBasic()` and `AuthBasic()` funcs.